meson.build

```diff
@@ -458,13 +476,14 @@ endif
 
 # Look for libsodium, an optional dependency.
 #--------------------------------------------------
-if (get_option('with_libsodium'))
+libsodium_dep_raw = cpp.find_library('sodium', required: get_option('with_libsodium'))
+if (libsodium_dep_raw.found())
     libsodium_dep = declare_dependency(
-        dependencies : cpp.find_library('libsodium', dirs : libdir),
+        dependencies : libsodium_dep_raw,
         link_args :  get_option('sodium_link_args'))
     config_h.set('HAVE_SODIUM', 1, description : 'Whether to use libsodium for cryptography.')
 else
-    libsodium_dep = declare_dependency()
+    libsodium_dep = dependency('', required: false)
 endif
```

Comment:
This is an overcomplication and makes the whole check messier (imo) with no real benefit

Reply:
Which part is?

Reply:
line 479,480,482. there is no need to declare the dependency twice in this way just to encompass the required inside the dependency. required defaults to true.